src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1345,6 +1345,42 @@ def check_user_in_second_cluster(self, cluster_url, cluster_token, username,
             logger.info("User %s not found yet, retrying in 5 seconds", username)
             time.sleep(5)
 
+class RgwRateLimit:
+    def get_rateLimit(self, scope: str, name: str ):
+        if scope=='user':
+            rate_limit_cmd = ['ratelimit', 'get', '--ratelimit-scope', scope, '--uid', name]
+        if scope=='bucket':
+            rate_limit_cmd = ['ratelimit', 'get', '--ratelimit-scope', scope, '--bucket', name]
+        try:
+            exit_code, out, err = mgr.send_rgwadmin_command(rate_limit_cmd)
+            if exit_code > 0:
+                raise DashboardException(f'Unable to get rate limit: {err}',
+                                            http_status_code=500, component='rgw')
+            return out
+        except SubprocessError as error:
+            raise DashboardException(error, http_status_code=500, component='rgw')
+
+    def set_rateLimit(self,scope:str,enabled,name, max_read_ops, max_write_ops,max_read_bytes,max_write_bytes):
+        if scope =='user':
+            rgw_rate_limit_cmd = ['ratelimit', 'set', '--ratelimit-scope', scope, '--uid', name, '--max-read-ops', max_read_ops,'--max-write-ops', max_write_ops,'--max-write-bytes',max_write_bytes, '--max-read-bytes',max_read_bytes]
+            rgw_rate_limit_enable_cmd= ['ratelimit', 'enable' if enabled=='true' else 'disable' , '--ratelimit-scope', scope, '--uid', name];
+
+        if scope=='bucket':
+            rgw_rate_limit_cmd = ['ratelimit', 'set', '--ratelimit-scope', scope, '--bucket', name, '--max-read-ops',max_read_ops, '--max-write-ops', max_write_ops,'--max-write-bytes',max_write_bytes, '--max-read-bytes',max_read_bytes ]
+            rgw_rate_limit_enable_cmd=['ratelimit', 'enable' if enabled=='true' else 'disable' , '--ratelimit-scope', scope, '--bucket', name];
```

Comment:
Same as above. We can improve readability be defining commands once

Reply:
fixed